From 30c639eccf27faecb2362848998b871e33d08197 Mon Sep 17 00:00:00 2001
From: Nisarg Pandya <nisarg.pandya@intel.com>
Date: Thu, 3 Sep 2015 10:16:59 -0700
Subject: [PATCH] Adding Support for D0 Silicon, The revision ID for D0 is
 different so needs to be added to the driver.

Change-Id: I77af2a9a4f5cd3b6a347daeac87077724a14b551
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5135
Signed-off-by: Nisarg Pandya <nisarg.pandya@intel.com>
---
 drivers/misc/heci/hw-ish.c          | 8 ++++++--
 drivers/misc/heci/pci-ish.c         | 4 +++-
 drivers/misc/heci/platform-config.h | 1 +
 3 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/heci/hw-ish.c b/drivers/misc/heci/hw-ish.c
index 3bb9799..caa7539 100644
--- a/drivers/misc/heci/hw-ish.c
+++ b/drivers/misc/heci/hw-ish.c
@@ -138,7 +138,9 @@ void ish_intr_enable(struct heci_device *dev)
 			(dev->pdev->revision & REVISION_ID_SI_MASK) ==
 			REVISION_ID_CHT_Bx_SI ||
 			(dev->pdev->revision & REVISION_ID_SI_MASK) ==
-			REVISION_ID_CHT_Kx_SI) {
+			REVISION_ID_CHT_Kx_SI ||
+			(dev->pdev->revision & REVISION_ID_SI_MASK) ==
+			REVISION_ID_CHT_Dx_SI) {
 		uint32_t	host_comm_val;
 
 		host_comm_val = ish_reg_read(dev, IPC_REG_HOST_COMM);
@@ -163,7 +165,9 @@ void ish_intr_disable(struct heci_device *dev)
 			(dev->pdev->revision & REVISION_ID_SI_MASK) ==
 			REVISION_ID_CHT_Bx_SI ||
 			(dev->pdev->revision & REVISION_ID_SI_MASK) ==
-			REVISION_ID_CHT_Kx_SI) {
+			REVISION_ID_CHT_Kx_SI ||
+			(dev->pdev->revision & REVISION_ID_SI_MASK) ==
+			REVISION_ID_CHT_Dx_SI) {
 		uint32_t	host_comm_val;
 
 		host_comm_val = ish_reg_read(dev, IPC_REG_HOST_COMM);
diff --git a/drivers/misc/heci/pci-ish.c b/drivers/misc/heci/pci-ish.c
index b2ac0ab..4e1bf46 100644
--- a/drivers/misc/heci/pci-ish.c
+++ b/drivers/misc/heci/pci-ish.c
@@ -896,7 +896,9 @@ static int ish_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 		(pdev->revision & REVISION_ID_SI_MASK) ==
 			REVISION_ID_CHT_Bx_SI ? "CHT B0" :
 		(pdev->revision & REVISION_ID_SI_MASK) ==
-			REVISION_ID_CHT_Kx_SI ? "CHT Kx/Cx" : "Unknown",
+			REVISION_ID_CHT_Kx_SI ? "CHT Kx/Cx" :
+		(pdev->revision & REVISION_ID_SI_MASK) ==
+			REVISION_ID_CHT_Dx_SI ? "CHT Dx" : "Unknown",
 		pdev->revision);
 #if defined(SUPPORT_Ax_ONLY)
 	pdev->revision = REVISION_ID_CHT_A0;
diff --git a/drivers/misc/heci/platform-config.h b/drivers/misc/heci/platform-config.h
index 933b8c7..2696bac 100644
--- a/drivers/misc/heci/platform-config.h
+++ b/drivers/misc/heci/platform-config.h
@@ -52,6 +52,7 @@
 #define	REVISION_ID_CHT_A0_SI	0x0
 #define	REVISION_ID_CHT_Bx_SI	0x10
 #define	REVISION_ID_CHT_Kx_SI	0x20
+#define	REVISION_ID_CHT_Dx_SI	0x30
 #define	REVISION_ID_CHT_B0	0xB0
 
 #define	REVISION_ID_SI_MASK	0x70
-- 
1.9.1

